### PR TITLE
SAM: support SAM CLI version 1.x

### PR DESCRIPTION
**Pull request by justinmk3**
## Testing:

- For each of SAM CLI 0.45.0, SAM CLI 1.0.0
  - For each of nodejs12.x, python3.7, dotnet2.1
    - Set the `aws.samcli.location` to point to new/old sam binary.
    - Verify that `Run Locally` works
    - Verify that `Debug Locally` works.

With sam 0.45.0, the Toolkit pulls these Docker images:

    Fetching lambci/lambda:dotnetcore2.1 Docker container image
    Fetching lambci/lambda:nodejs12.x Docker container image
    Fetching lambci/lambda:python3.7 Docker container image

With sam 1.0.0, the Toolkit pulls these Docker images:

    Fetching amazon/aws-sam-cli-emulation-image-dotnetcore2.1:latest Docker container image
    Fetching amazon/aws-sam-cli-emulation-image-nodejs10.x:latest Docker container image
    Fetching amazon/aws-sam-cli-emulation-image-nodejs12.x:latest Docker container image
    Fetching amazon/aws-sam-cli-emulation-image-python3.7:latest Docker container image

Per https://github.com/awslabs/aws-sam-cli/pull/2066/ , the new docker images are named like:

    amazon/aws-sam-cli-build-image-<runtime>
    amazon/aws-sam-cli-emulation-image-<runtime>

## Notes:

- Elected not to include `justMyCode:false` in the dotnet config.
- In a test of nodejs12.x, saw this error (transient, went away after retry):
  ```
  docker.errors.ImageNotFound: 404 Client Error: Not Found ("No such image: amazon/aws-sam-cli-emulation-image-nodejs12.x:latest")
  ```
- ref https://github.com/aws/aws-toolkit-jetbrains/pull/1886
- ref e1e6e5f814ce251ef2eefa1ac958fb2e92dd9c20
- ref 60a2eab4d1ada8e52f2ed4b52305c5e69aad7f3e

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by hunterwerlla** [comment]
Does everything work as expected? (including .net?)

File: src/lambda/local/debugConfiguration.ts [hunk 1]
```diff
@@ -80,7 +80,10 @@ export function makeCoreCLRDebugConfiguration({
         name: 'SamLocalDebug',
         type: 'coreclr',
         request: 'attach',
-        processId: '1',
+        // Since SAM CLI 1.0 we cannot assume PID=1. So use processName=dotnet
```
Comment (awschristou) [inline src/lambda/local/debugConfiguration.ts:83]: Are the changes to this file backwards compatible with using SAM CLI versions earlier than 1.0?

Comment (justinmk3) [reply src/lambda/local/debugConfiguration.ts:83]: With sam `0.45.0`, using `Debug Locally` pulls Docker image `lambci/lambda:dotnetcore2.1`, the debugger attaches and breakpoints are successfully hit in vscode.

**Comment by justinmk3** [comment]
> Does everything work as expected? (including .net?)

Performed tests for sam `0.45` and `1.0` with node10, node12, python3.7, dotnet2.1. Added details to PR description and commit message .

**Comment by justinmk3** [comment]
artifact: [aws-toolkit-vscode-1.11.0-SNAPSHOT.vsix.zip](https://github.com/aws/aws-toolkit-vscode/files/4861210/aws-toolkit-vscode-1.11.0-SNAPSHOT.vsix.zip)

**Comment by bryceitoc9** [comment]
Confirming that SAM 0.51.0 still pulls the old images. Run works for all runtimes (including C#).

**Pull request merged**